File: backend/party-server/src/main/java/io/featurehub/party/Application.java
package io.featurehub.party;

import cd.connect.jersey.JerseyHttp2Server;
import cd.connect.jersey.common.CorsFilter;
import cd.connect.jersey.common.LoggingConfiguration;
import cd.connect.lifecycle.ApplicationLifecycleManager;
import cd.connect.lifecycle.LifecycleStatus;
import cd.connect.openapi.support.ReturnStatusContainerResponseFilter;
import io.featurehub.dacha.DachaFeature;
import io.featurehub.edge.EdgeFeature;
import io.featurehub.edge.EdgeResourceFeature;
import io.featurehub.health.MetricsHealthRegistration;
import io.featurehub.jersey.config.CommonConfiguration;
import io.featurehub.jersey.config.EndpointLoggingListener;
import io.featurehub.mr.ManagementRepositoryFeature;
import io.featurehub.mr.utils.NginxUtils;
import io.featurehub.publish.NATSFeature;
import io.featurehub.web.security.oauth.OAuth2Feature;
import org.glassfish.jersey.server.ResourceConfig;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class Application {
  private static final Logger log = LoggerFactory.getLogger(io.featurehub.Application.class);

  public static void main(String[] args) {
    System.setProperty("user.timezone", "UTC");
    try {
      NginxUtils.seeIfWeNeedToRunNginx();
      new io.featurehub.party.Application().run();
    } catch (Exception e) {
      log.error("failed", e);
      System.exit(-1);
    }
  }


  private void run() throws Exception {
    // register our resources, try and tag them as singleton as they are instantiated faster
    ResourceConfig config = new ResourceConfig(
      CommonConfiguration.class,
      LoggingConfiguration.class,
      ReturnStatusContainerResponseFilter.class,
      EndpointLoggingListener.class,
      NATSFeature.class,
      CorsFilter.class,
      OAuth2Feature.class,
      ManagementRepositoryFeature.class,
      EdgeResourceFeature.class,
      EdgeFeature.class,
      DachaFeature.class
      );

    // check if we should list on a different port
    MetricsHealthRegistration.Companion.registerMetrics(config);

    new JerseyHttp2Server().start(config);

    log.info("MR Launched - (HTTP/2 payloads enabled!)");

    // tell the App we are ready
    ApplicationLifecycleManager.updateStatus(LifecycleStatus.STARTED);

    Thread.currentThread().join();
  }
}

File: backend/party-server-ish/src/main/kotlin/io.featurehub.party/Application.kt
package io.featurehub.party

import cd.connect.app.config.ConfigKey
import cd.connect.app.config.DeclaredConfigResolver
import cd.connect.jersey.common.CorsFilter
import io.featurehub.edge.EdgeGetFeature
import io.featurehub.health.MetricsHealthRegistration.Companion.registerMetrics
import io.featurehub.jersey.FeatureHubJerseyHost
import io.featurehub.lifecycle.TelemetryFeature
import io.featurehub.mr.ManagementRepositoryFeature
import io.featurehub.publish.ChannelConstants
import io.featurehub.rest.Info
import io.featurehub.web.security.oauth.OAuth2Feature
import org.glassfish.jersey.server.ResourceConfig
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Application {
  private val log = LoggerFactory.getLogger(io.featurehub.Application::class.java)

  @ConfigKey("cache.name")
  var name = ChannelConstants.DEFAULT_CACHE_NAME

  init {
    DeclaredConfigResolver.resolve(this)
  }

  @Throws(Exception::class)
  private fun run() {
    log.info("starting party-server-ish, wish me luck!")

    // register our resources, try and tag them as singleton as they are instantiated faster
    val config = ResourceConfig(
      CorsFilter::class.java,
      OAuth2Feature::class.java,
      ManagementRepositoryFeature::class.java,
      EdgeGetFeature::class.java,
      TelemetryFeature::class.java,
    )

      // check if we should list on a different port
    registerMetrics(config)
    FeatureHubJerseyHost(config).start()
    log.info("Party-Server-ish Launched - (HTTP/2 payloads enabled!)")

    Thread.currentThread().join()
  }

  companion object {
    private val log: Logger = LoggerFactory.getLogger(Application::class.java)

    @JvmStatic
    fun main(args: Array<String>) {
      System.setProperty("user.timezone", "UTC")
      System.setProperty(Info.APPLICATION_NAME_PROPERTY, "party-server-ish")

      System.clearProperty("nats.urls")

      try {
        Application().run()
      } catch (e: Exception) {
        log.error("failed", e)
        System.exit(-1)
      }
    }
  }
}
